### PR TITLE
feature/registration-regression-fix: Registration was NOT working as …

### PR DESCRIPTION
**Pull request by AltusBarry**
…expected when no redirects were present

### CODE REVIEW

File: authentication_service/views.py [hunk 1]
```diff
@@ -239,6 +240,10 @@ def form_valid(self, form):
                 data["language_code"] = self.language
                 question = models.UserSecurityQuestion.objects.create(**data)
 
+        # If get_success_url already returns a response object, immediately
+        # return it.
+        if not isinstance(self.get_success_url(), str):
```
Comment (cobusc) [inline authentication_service/views.py]: I'd prefer if the function was only called once.

Comment (cobusc) [reply authentication_service/views.py]: Also, if the response is a `str` then what does that mean? Do we just ignore the value?

Comment (AltusBarry) [reply authentication_service/views.py]: The issue is, get_success_url should not return anything other than a str. Te more explicit check would be `if isinstance(self.get_success_url(), HttpResponse)`.

Comment (cobusc) [reply authentication_service/views.py]: I see. So it is the fact that `get_success_url` has been overloaded to do a `render()` as well.
Will we then ever have a use for `response`? Should we then have
```
result = self.get_success_url()
if isinstance(result, HttpResponse):
    return result
return redirect(result)
```

**Pull request closed without merging**